cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/aws/AwsClusterSettingsStep.tsx

```diff
@@ -74,128 +71,130 @@ function yupStringRequired(errorMessage: string) {
-    };
-
-    const onClusterNameChange = (clusterName: string) => {
-        onFieldChange(clusterName, 'CLUSTER_NAME');
-    };
-
-    const onOsImageSelected = (clusterName: string) => {
-        onFieldChange(clusterName, 'IMAGE_INFO');
+        onFieldChange(AWS_FIELDS.NODE_PROFILE, instanceType);
     };
 
     return (
-        <div className="cluster-settings-container" cds-layout="m:lg">
-            <h3>Management Cluster settings</h3>
-            <div cds-layout="grid gap:m" key="section-holder">
-                <div cds-layout="col:6" key="cluster-name-section">
-                    <ClusterName
-                        field={'CLUSTER_NAME'}
-                        errors={errors}
-                        register={register}
-                        clusterNameChange={onClusterNameChange}
-                        placeholderClusterName={'my-aws-cluster'}
-                    />
-                </div>
-                <div cds-layout="col:6" key="instance-type-section">
-                    <NodeProfile
-                        field={'NODE_PROFILE'}
-                        nodeInstanceTypes={nodeInstanceTypes}
-                        errors={errors}
-                        register={register}
-                        nodeInstanceTypeChange={onInstanceTypeChange}
-                        selectedInstanceId={selectedInstanceTypeId}
-                    />
-                </div>
-                <div cds-layout="col:6">
-                    <PageNotification notification={notification} closeCallback={dismissAlert}></PageNotification>
-                </div>
-                <div cds-layout="col:12">
-                    <OsImageSelect
-                        osImageTitle={'Amazon Machine Image(AMI)'}
-                        images={images}
-                        field={'IMAGE_INFO'}
-                        errors={errors}
-                        register={register}
-                        onOsImageSelected={onOsImageSelected}
-                    />
-                </div>
-
-                <div cds-layout="grid col:12 p-t:lg">
-                    <CdsButton cds-layout="col:start-1" status="success" onClick={handleSubmit(handleMCCreation)}>
-                        <CdsIcon shape="cluster" size="sm"></CdsIcon>
-                        Create Management cluster
-                    </CdsButton>
-                    <CdsButton cds-layout="col:end-12" action="flat">
-                        View configuration details
-                    </CdsButton>
+        <FormProvider {...methods}>
+            <div className="cluster-settings-container" cds-layout="m:lg">
+                <h3>Management Cluster settings</h3>
+                <div cds-layout="grid gap:m" key="section-holder">
+                    <div cds-layout="col:4" key="cluster-name-section">
+                        <ClusterName
+                            field={AWS_FIELDS.CLUSTER_NAME}
+                            clusterNameChange={(value) => {
```

Comment:
One of the reasons for the original cluster name interface to send back the field name was to allow the caller to re-use a generic handler. In this case, you would be able to use onFieldChange directly, rather than having to create this closure on the fly. It might be worth talking over.

Reply:
Yes. let's find a time to discuss it.